workers/github_worker_base.rb

```diff
@@ -0,0 +1,112 @@
+require_relative "worker_base"
+require_relative "../shared/models/provider_credential"
+require_relative "../shared/logging_module"
+require_relative "../services/test_runner_service"
+require_relative "../services/code_hosting/git_hub_service"
+
+module FastlaneCI
+  # Base class for GitHub workers
+  class GithubWorkerBase < WorkerBase
+    include FastlaneCI::Logging
+
+    attr_accessor :provider_credential
+    attr_accessor :project
+    attr_accessor :github_service
+    attr_accessor :serial_task_queue
+    attr_accessor :project_full_name
+    attr_accessor :target_branches_set
+    attr_writer :git_repo
+
+    def provider_type
+      return FastlaneCI::ProviderCredential::PROVIDER_CREDENTIAL_TYPES[:github]
+    end
+
+    def initialize(provider_credential: nil, project: nil)
+      self.provider_credential = provider_credential
+      self.github_service = FastlaneCI::GitHubService.new(provider_credential: provider_credential)
+      self.project = project
+
+      self.target_branches_set = Set.new
+      project.job_triggers.each do |trigger|
+        if trigger.type == self.trigger_type
+          self.target_branches_set.add(trigger.branch)
+        end
+      end
+
+      self.project_full_name = project.repo_config.git_url
+
+      if project.repo_config.kind_of?(FastlaneCI::GitRepoConfig)
+        self.project_full_name = project.repo_config.full_name unless project.repo_config.full_name.nil?
+      end
+
+      self.serial_task_queue = TaskQueue::TaskQueue.new(name: "#{self.project_full_name}:#{provider_credential.ci_user.email}")
+
+      super() # This starts the work by calling `work`
+    end
+
+    def thread_id
+      if @thread_id
+        return @thread_id
+      end
+
+      time_nano = Time.now.nsec
+      @thread_id = "GithubWorker:#{time_nano}: #{self.serial_task_queue.name}"
```

Comment:
Let's use the current class name instead of `GithubWorker` so the logging is more precise

Reply:
done